test/api_saveProduct_test.dart

```diff
@@ -487,4 +488,39 @@ void main() {
         // some tests can be slow here
         Duration(seconds: 90),
       ));
+
+  group('No nutrition data', () {
+    test('No nutrition data with nutriments', () async {
+      Product product = Product(
+          noNutritionData: true,
+          nutriments: Nutriments(
+            salt: 1.0,
+          ));
+
+      expect(product.noNutritionData, isTrue);
+      expect(product.nutriments, isNull);
+
+      var serverData = product.toServerData();
+      expect(serverData[ProductField.NO_NUTRITION_DATA.key],
+          equals(JsonHelper.checkboxOnValue));
+      expect(serverData[ProductField.NUTRIMENTS.key], equals('{}'));
```

Comment:
It's a bit too much: all we need are the `noNutritionData` and `nutriments` fields.